### PR TITLE
Update cache action (add `~/.cabal-devx` and `~/.cache` to the cache)

### PR DESCRIPTION
**Pull request by yvan-sraka**
To test this action, after having it merged, we could re-run this workflow https://github.com/IntersectMBO/cardano-base/actions/workflows/haskell.yml

### CODE REVIEW

File: cache/action.yml [hunk 1]
```diff
@@ -8,9 +8,24 @@ inputs:
 runs:
   using: "composite"
   steps:
+    - name: Bootstrap devx shell
+      uses: input-output-hk/actions/devx@latest
+      with:
+        platform: 'x86_64-linux'
+        target-platform: ''
+        compiler-nix-name: ${{ inputs.ghc_version }}
```
Comment (yvan-sraka) [inline cache/action.yml:16]: I should have called `ghc_version` rather `compiler-nix-name` ... if I rename it I should also change it here https://github.com/IntersectMBO/cardano-base/blob/master/.github/workflows/haskell.yml#L158

Comment (yvan-sraka) [reply cache/action.yml:16]: (or maybe the action should takes as input also `platform`, `target-platform`, `minimal` and `iog`? even if all GitHub Codespace needs HLS, runs on `x86_64-linux` and likely want to offers `-iog` flavor)

Comment (angerman) [reply cache/action.yml:16]: for now let's try to keep it simple and build a reliable PoC, then expand.

We won't do target platforms yet, flavours _could_ be an issue, but as long as we keep the one in CI and the one in the devcontainer the same it should be ok.

File: cache/action.yml [hunk 2]
```diff
@@ -8,9 +8,24 @@ inputs:
...
+        cabal update
+        timeout 15m haskell-language-server --project-ghc-version
     - name: Cache build artifacts (used to speed up GitHub Codespaces bootstrapping)
       uses: actions/upload-artifact@v4
       with:
         name: cache-${{ github.event.pull_request.head.sha || github.sha }}-${{ inputs.ghc_version }}
         path: |
           ./dist-newstyle
+          $HOME/.cabal-devx
+          $HOME/.cache
```
Comment (angerman) [inline cache/action.yml:32]: we might not need .cache, if you are concerned it breaks HLS too easily, but we can try.

**Review by angerman — APPROVED** [review]

**Pull request merged**